### PR TITLE
feat:show table in mobile with hscroll

### LINKED ISSUE #969: [Feature request]: Show who created a booking on the booking page
### Contact Details

_No response_

### Is your feature request related to a problem? Please describe?

I am on a mobile phone. I open a booking and I see an issue. 

On desktop, when I go to the booking page. I have a column that shows me who created a booking. That way I can get in touch with this person to discuss things around the booking. 

I have no way to do this on mobile. 

### Describe the solution you'd like

Add the booking creator data on the booking page on mobile and desktop. 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rajeshj11**
/closes #969 
showing the table and providing the horizontal scroll if any overflow is there.

Impact: mobile device will able to view the table data.

### CODE REVIEW

**Comment by DonKoko** [comment]
@carlosvirreira Rajesh has taken on this issue. He has updated all tables in the platform to make them horizontally scrollable so users can see full data.
I am deploying it to staging: https://github.com/Shelf-nu/shelf.nu/actions/runs/10490727074

Lets do some testing to make sure everything is okey.

**Review by DonKoko — CHANGES_REQUESTED** [review]
Writing some feedback, based on what we discussed:

- [x] Some `<Td>` components have `className="table-cell"`. This does nothing as they are already table cells. Lets remove it
- [x] With the new approach the name column(first column) should not grow anymore(only mobile). It should just shrink to the content. Basically default table functionality. This is the case on some pages (assets) but not the case on the bookings index. Should be consistent. Also please check all indecies because in the categories index, for example, the description column grows. Again we dont want that on mobile, just desktop.
- [x] The gradient is a bit too thin(change to 35px). Also the gradient at the left point has to be 100% transparent so we have a smooth transition. Now it looks like a cut-off line
- [x] The gradient should only be there if the table is overflowing horizontally(both desktop and mobile). For example in the Non-registered members index, there are very few columns so they fit and there is no horizontal scroll, however there is still a gradient covering the right column
- [ ] team/users index still has the chevrons, even tho it has the new style of table. Needs to be adjusted
- [ ] Booking/manage-assets modal is basically broken now. I cant scroll it or view it. Same for location/manage-assets modal.

**Comment by carlosvirreira** [comment]
- [ ] Batch Action is broken on Mobile (Safari) - On Asset Index it works well

Perhaps adjacently related:

On asset dashboard - some widgets seem to assume they are side-scroll-able and therefore show some sort of shadow on the right side.

Same as locations on mobile. Action panel looks broken. 

Calendar component also seems to be cracking.

User table on mobile seems to also show the right pointing chevron to show its click-able but also sidescroll.

@DonKoko my comments.

**Comment by rajeshj11** [comment]
> Writing some feedback, based on what we discussed:
> 
> * [ ]  Some `<Td>` components have `className="table-cell"`. This does nothing as they are already table cells. Lets remove it
> * [ ]  With the new approach the name column(first column) should not grow anymore(only mobile). It should just shrink to the content. Basically default table functionality. This is the case on some pages (assets) but not the case on the bookings index. Should be consistent. Also please check all indecies because in the categories index, for example, the description column grows. Again we dont want that on mobile, just desktop.
> * [ ]  The gradient is a bit too thin(change to 35px). Also the gradient at the left point has to be 100% transparent so we have a smooth transition. Now it looks like a cut-off line
> * [ ]  The gradient should only be there if the table is overflowing horizontally(both desktop and mobile). For example in the Non-registered members index, there are very few columns so they fit and there is no horizontal scroll, however there is still a gradient covering the right column
> * [ ]  team/users index still has the chevrons, even tho it has the new style of table. Needs to be adjusted
> * [ ]  Booking/manage-assets modal is basically broken now. I cant scroll it or view it. Same for location/manage-assets modal.

resolved all the issues. i am not 100% sure about the gradient(CSS background thing. i might need help on that).

**Comment by rajeshj11** [comment]
> * [ ]  Batch Action is broken on Mobile (Safari) - On Asset Index it works well
> 
> Perhaps adjacently related:
> 
> On asset dashboard - some widgets seem to assume they are side-scroll-able and therefore show some sort of shadow on the right side.
> 
> Same as locations on mobile. Action panel looks broken.
> 
> Calendar component also seems to be cracking.
> 
> User table on mobile seems to also show the right pointing chevron to show its click-able but also sidescroll.
> 
> @DonKoko my comments.
> 
> IMG_7759

I am not sure about the calendar thing. @DonKoko I have checked with main branch it is same. other issues were taken care.

**Review by DonKoko — CHANGES_REQUESTED** [review]
- [ ] with the new approach, the description column on categories index should not be truncated in 1 row. Can you make it that it takes 3 rows. We have a function alreayd in shelf that counts the rows and truncates the text

For the gradient do the following:
1. Increase width to 35px
2. Set the gradient to : `linear-gradient(to right, rgba(255, 255, 255, 0) 0%, #fff 100%)`

**Comment by rajeshj11** [comment]
> * [ ]  with the new approach, the description column on categories index should not be truncated in 1 row. Can you make it that it takes 3 rows. We have a function alreayd in shelf that counts the rows and truncates the text
> 
> For the gradient do the following:
> 
> 1. Increase width to 35px
> 2. Set the gradient to : `linear-gradient(to right, rgba(255, 255, 255, 0) 0%, #fff 100%)`

ack

**Comment by carlosvirreira** [comment]
What I meant about the calendar.

IMG_7762

Besides that, looks good to me (safari, iPhone)

File: app/components/table.tsx [hunk 1]
```diff
@@ -9,10 +9,49 @@ export function Table({
...
+      if (container) {
+        const hasOverflow = container.scrollWidth > container.clientWidth;
+        const hasReachedEnd =
+          container.scrollLeft + container.clientWidth >= container.scrollWidth;
+        setIsOverflowing(hasOverflow && !hasReachedEnd);
+      }
+    };
+
+    checkOverflow(); // Initial check
+    window.addEventListener("resize", checkOverflow); // Check on resize
```
Comment (DonKoko) [inline app/components/table.tsx]: If you want to do this, inside the useEffect you should have a `if (window)`

File: app/components/table.tsx [hunk 2]
```diff
@@ -9,10 +9,49 @@ export function Table({
   children: React.ReactNode;
   className?: string;
 }) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [isOverflowing, setIsOverflowing] = useState(false);
+
+  useEffect(() => {
```
Comment (DonKoko) [inline app/components/table.tsx]: Can you please abstract this into its own hook called `useTableIsOverflowing` which returns the value?

**Review by DonKoko — CHANGES_REQUESTED** [review]

**Review by DonKoko — APPROVED** [review]

**Pull request merged**